src/controllers/userController.js

```diff
@@ -112,11 +114,61 @@ const updateUser = async (req, res) => {
   }
 };
 
+
+const searchUsers = async (req, res) => {
+  try {
+    const { query, id, page = 1, limit = 10 } = req.query;
+
+
+    if (!query || typeof query !== "string" || query.length > 100) {
+      return res.status(400).json({ error: "Invalid query parameter." });
+    }
+
+    const safeQuery = escapeWildcards(query);
+
+    const offset = (page - 1) * limit;
+
+    const where = {};
+    if (query) where.username = { [Op.like]: `${safeQuery}%` };
+    if (id) where.id = id;
+
+
+    const users = await User.findAndCountAll({
+      where,
+      attributes: ["id", "username", "email", "birthday"],
+      limit: parseInt(limit),
+      offset: parseInt(offset),
+    });
+
+    if (users.rows.length === 0) {
+      return res.status(404).json({
+        message: "No users found matching the given criteria.",
+        criteria: { query, id },
+      });
+    }
+
+    res.status(200).json({
+      metadata: {
+        query,
+        caseSensitive: false,
+        totalResults: users.count,
+        totalPages: Math.ceil(users.count / limit),
+        currentPage: parseInt(page),
+      },
+      data: users.rows,
+    });
```

Comment:
Please confider excluding user that searching for other users from the result list